test/openssl/test_pkey_rsa.rb

```diff
@@ -15,7 +15,15 @@ def test_no_private_exp
 
   def test_private
     # Generated by key size and public exponent
-    key = OpenSSL::PKey::RSA.new(512, 3)
+    #
+    # SP800 requires ossl_ifc_ffc_compute_security_bits that the return value
+    # (strength in bits) is more than equal RSA_FIPS1864_MIN_KEYGEN_STRENGTH
+    # (112) in FIPS.
+    # https://github.com/openssl/openssl/blob/3c6e11495975a4eda4cc5886080afed6203711ac/crypto/rsa/rsa_sp800_56b_gen.c#L176-L182
+    # The ossl_ifc_ffc_compute_security_bits returns 112 with the argument nbits
+    # 2048.
+    # https://github.com/openssl/openssl/blob/3c6e11495975a4eda4cc5886080afed6203711ac/crypto/rsa/rsa_lib.c#L334-L335
+    key = OpenSSL::PKey::RSA.new(2048, 65537)
     assert(key.private?)
```

Comment:
By the way, it probably should have been `assert_equal(true, [...])`  rather than the bare `assert()`.

Reply:
> Since generating a 2048-bit RSA key takes non-negligible amount of time, I don't want to do it more than strictly necessary. Perhaps we can move this assertion to `test_new`?

Did you mean that we keep the `key = OpenSSL::PKey::RSA.new(512, 3)` in `test_private`, only executing the `test_private` in non-FIPS cases?

In the `test_new`, there is already `key2 = OpenSSL::PKey::RSA.new(2048, 65537)`. So, I think we don't need to change the current `test_new` on this PR. Is my understanding right?

Reply:
> By the way, it probably should have been `assert_equal(true, [...])` rather than the bare `assert()`.

Sure, that seems more accurate.

Reply:
> > Since generating a 2048-bit RSA key takes non-negligible amount of time, I don't want to do it more than strictly necessary. Perhaps we can move this assertion to `test_new`?
> 
> Did you mean that we keep the `key = OpenSSL::PKey::RSA.new(512, 3)` in `test_private`, only executing the `test_private` in non-FIPS cases?
> 
> In the `test_new`, there is already `key2 = OpenSSL::PKey::RSA.new(2048, 65537)`. So, I think we don't need to change the current `test_new` on this PR. Is my understanding right?

I understand how to modify the `test_new` is related to the following comment of yours at another part. So, I will move the `OpenSSL::PKey::RSA.new(2048, 65537)` to the `test_new`.

> The purpose of this assertion is to check that the second parameter of `RSA.new` correctly changes the public exponent which defaults to 65537.
> 
> We should not change it. Please instead skip if it is not allowed in the FIPS mode.

Reply:
I don't want `test_private` to generate a 2048-bit RSA key just to assert `key.private? == true` since it can take 0.5s on a slow machine. It's not that significant, but I try to avoid writing slow tests because they can pile up quickly.

It does make sense to check `rsa.private?` returns true on a newly generated RSA key though, but we can do it in `test_new` where we have to generate it anyway. I don't think we have to keep the 512-bit key test in `test_private` if we do so.

Reply:
> By the way, it probably should have been `assert_equal(true, [...])` rather than the bare `assert()`.

I want to fix this on another PR. Because the case may not be not only the `test_private` in the `test_pkey_rsa.rb`.

Reply:
I updated the `test_new` with the `OpenSSL::PKey::RSA.new(2048, 65537)`.